api/types/backend/backend.go

```diff
@@ -98,6 +99,9 @@ type ExecProcessConfig struct {
 type ContainerCommitConfig struct {
 	types.ContainerCommitConfig
 	Changes []string
+	// TODO: ContainerConfig is only used by the dockerfile Builder, so remove it
+	// once the Builder has been updated to use a different interface
+	ContainerConfig *container.Config
```

Comment:
Should we mark in the name that this should not be used by the API so we don't need to deprecate it? Or maybe it could be useful for `docker commit` as well.